### PR TITLE
Update material and cupertino localizations

### LINKED ISSUE #130874: Incorrect "Replacements not found" translations for ru language
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Create a sample project with textfield with spellCheckConfiguration: SpellCheckConfiguration()
2. Run project on ios
3. Add the word "Vitamin" (for example)
4. Select this word
5. Popup with the text "No replacements Found" is appears

### Expected results

In Russian localization, there must be a text - "Замен не найдено",

### Actual results

This text is equal to "No replacements found",

### Code sample

<details><summary>Code sample</summary>

```dart
    TextField(spellCheckConfiguration: SpellCheckConfiguration());
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

Screenshot 2023-07-19 at 15 03 50

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4 22F66 darwin-x64, locale en-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.80.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by thkim1011**
See title.

Fixes https://github.com/flutter/flutter/issues/130874.

### CODE REVIEW

**Comment by HansMuller** [comment]
Should include a test that checks for "noSpellCheckReplacementsLabel" translations.

**Comment by justinmc** [comment]
@itsjustkevin What do you think about cherry picking this into stable?  Lots of files changed but it's just translations, have we done a translation cherry pick before?

Currently on stable, iOS users with any text field with spell check enabled will see "No replacements found" in English (untranslated) for words that don't have suggestions.  Now that I type that out it seems likely to be a small portion of our users (1. iOS 2. spell check enabled 3. user clicking a word with no suggestions).  But people are apparently encountering it since https://github.com/flutter/flutter/issues/130874 was filed.

File: packages/flutter_localizations/lib/src/l10n/cupertino_ar.arb [hunk 1]
```diff
@@ -42,6 +42,6 @@
   "tabSemanticsLabel": "علامة التبويب $tabIndex من $tabCount",
   "modalBarrierDismissLabel": "رفض",
   "searchTextFieldPlaceholderLabel": "بحث",
-  "noSpellCheckReplacementsLabel": "No Replacements Found",
+  "noSpellCheckReplacementsLabel": "لم يتم العثور على بدائل.",
```
Comment (hadysata) [inline packages/flutter_localizations/lib/src/l10n/cupertino_ar.arb:45]:
```suggestion
  "noSpellCheckReplacementsLabel": "لم يتم العثور على بدائل",
```
No need for this extra dot(.) in the Arabic translations

This is from Native IOS:
IMG_4702

Comment (thkim1011) [reply packages/flutter_localizations/lib/src/l10n/cupertino_ar.arb:45]: I requested a change to the translations internally.

File: packages/flutter_localizations/lib/src/l10n/cupertino_pt.arb [hunk 2]
```diff
@@ -18,10 +18,10 @@
   "cutButtonLabel": "Cortar",
   "copyButtonLabel": "Copiar",
   "pasteButtonLabel": "Colar",
-  "selectAllButtonLabel": "Selecionar tudo",
+  "selectAllButtonLabel": "Selecionar Tudo",
```
Comment (tgucio) [inline packages/flutter_localizations/lib/src/l10n/cupertino_pt.arb:21]: "tudo" shouldn't be capitalised here.

File: packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb [hunk 3]
```diff
@@ -28,10 +28,10 @@
   "cutButtonLabel": "Wytnij",
   "copyButtonLabel": "Kopiuj",
   "pasteButtonLabel": "Wklej",
-  "selectAllButtonLabel": "Zaznacz wszystko",
+  "selectAllButtonLabel": "Wybierz wszystkie",
```
Comment (tgucio) [inline packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb:31]: This is incorrect. "Wybierz wszystko" refers to selecting all [countable] elements. This string is used for text selection though which should keep the existing "Zaznacz wszystko" translation. I am suspecting there could be multiple other languages with a similar translation error.

Maybe we could add a description to `selectAllButtonLabel` in _en.arb to explain what it's used for?

File: packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb [hunk 4]
```diff
@@ -28,10 +28,10 @@
...
   "tabSemanticsLabel": "Karta $tabIndex z $tabCount",
   "modalBarrierDismissLabel": "Zamknij",
   "searchTextFieldPlaceholderLabel": "Szukaj",
-  "noSpellCheckReplacementsLabel": "Nie znaleziono zastąpień",
+  "noSpellCheckReplacementsLabel": "Brak wyników zamieniania",
```
Comment (tgucio) [inline packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb:35]: This should be left as is, source: Apple's macOS glossary:

```
    <TextItem>
      <Description> No Replacements Found label in menu bar. Shown when the spellchecker cannot offer any guesses for a given word. </Description>
      <Position>No Replacements Found</Position>
      <TranslationSet>
        <base loc="en">No Replacements Found</base>
        <tran loc="pl">Nie znaleziono zastąpień</tran>
      </TranslationSet>
    </TextItem>
```

IMHO for Cupertino translations, the localised strings should be checked with Apple glossaries (https://developer.apple.com/download/all/?q=glossaries).

<details>
  <summary>"No Replacements Found" translations from Apple glossaries</summary>

```
% grep -r "<base loc=\"en\">No Replacements Found</base>" -A 1 | grep -v -- "^--$" | grep -v loc=\"en\" | sort
./Arabic/UIKitCore.lg-        <tran loc="ar">لم يتم العثور على بدائل</tran>
./Brazilian/UIKitCore.lg-        <tran loc="pt-BR">Nenhuma Substituição Encontrada</tran>
./Canadian French/UIKitCore.lg-        <tran loc="fr-CA">Aucune proposition</tran>
./Catalan/UIKitCore.lg-        <tran loc="ca">No s’han trobat substituts</tran>
./Croatian/UIKitCore.lg-        <tran loc="hr">Zamjene nisu nađene</tran>
./Czech/UIKitCore.lg-        <tran loc="cs">Nenalezeny žádné náhrady</tran>
./Danish/UIKitCore.lg-        <tran loc="da">Ingen erstatninger fundet</tran>
./Dutch/UIKitCore.lg-        <tran loc="nl">Geen vervanging gevonden</tran>
./Finnish/UIKitCore.lg-        <tran loc="fi">Korvaavuuksia ei löytynyt</tran>
./French/UIKitCore.lg-        <tran loc="fr">Aucune proposition</tran>
./German/UIKitCore.lg-        <tran loc="de">Kein Ersatz gefunden</tran>
./Greek/UIKitCore.lg-        <tran loc="el">Δεν βρέθηκαν αντικαταστάσεις</tran>
./Hebrew/UIKitCore.lg-        <tran loc="he">לא נמצאו חלופות</tran>
./Hindi/UIKitCore.lg-        <tran loc="hi">कोई प्रतिस्थापन सुझाव नहीं मिला</tran>
./Hong Kong/UIKitCore.lg-        <tran loc="zh-HK">找不到轉換字詞</tran>
./Hungarian/UIKitCore.lg-        <tran loc="hu">Nem található csere</tran>
./Indonesian/UIKitCore.lg-        <tran loc="id">Pengganti Tidak Ditemukan</tran>
./Italian/UIKitCore.lg-        <tran loc="it">Nessuna sostituzione trovata</tran>
./Japanese/UIKitCore.lg-        <tran loc="ja">代替候補が見つかりません</tran>
./Korean/UIKitCore.lg-        <tran loc="ko">발견된 대치 항목 없음</tran>
./Latin American/UIKitCore.lg-        <tran loc="es-419">No se encontró ninguna palabra de sustitución</tran>
./Malay/UIKitCore.lg-        <tran loc="ms">Tiada Pengganti Ditemui</tran>
./Norwegian Bokmal/UIKitCore.lg-        <tran loc="no">Fant ingen erstatninger</tran>
./Polish/UIKitCore.lg-        <tran loc="pl">Nie znaleziono zastąpień</tran>
./Portuguese/UIKitCore.lg-        <tran loc="pt-PT">Nenhuma substituição encontrada</tran>
./Romanian/UIKitCore.lg-        <tran loc="ro">Nicio înlocuire găsită</tran>
./Russian/UIKitCore.lg-        <tran loc="ru">Замен не найдено</tran>
./Simplified Chinese/UIKitCore.lg-        <tran loc="zh-CN">未找到替换文本</tran>
./Slovak/UIKitCore.lg-        <tran loc="sk">Nenašli sa žiadne náhrady</tran>
./Spanish/UIKitCore.lg-        <tran loc="es">No se ha encontrado ninguna palabra de sustitución</tran>
./Swedish/UIKitCore.lg-        <tran loc="sv">Inga ersättningar hittades</tran>
./Thai/UIKitCore.lg-        <tran loc="th">ไม่พบการแทนที่</tran>
./Traditional Chinese/UIKitCore.lg-        <tran loc="zh-TW">找不到替代文字</tran>
./Turkish/UIKitCore.lg-        <tran loc="tr">Hiçbir Değişim Bulunamadı</tran>
./Ukrainian/UIKitCore.lg-        <tran loc="uk">Замін не знайдено</tran>
./Vietnamese/UIKitCore.lg-        <tran loc="vi">Không tìm thấy mục thay thế</tran>
```
</details>

File: packages/flutter_localizations/lib/src/l10n/material_ar.arb [hunk 5]
```diff
@@ -146,11 +146,11 @@
   "bottomSheetLabel": "بطاقة سفلية",
   "scrimOnTapHint": "إغلاق \"$modalRouteContentName\"",
   "keyboardKeyShift": "Shift",
-  "expansionTileExpandedHint": "double tap to collapse'",
-  "expansionTileCollapsedHint": "double tap to expand",
-  "expansionTileExpandedTapHint": "Collapse",
-  "expansionTileCollapsedTapHint": "Expand for more details",
-  "expandedHint": "Collapsed",
-  "collapsedHint": "Expanded",
+  "expansionTileExpandedHint": "يُرجى النقر مرّتين للتصغير.",
```
Comment (tgucio) [inline packages/flutter_localizations/lib/src/l10n/material_ar.arb:149]: Remove final full stop here and elsewhere.

File: packages/flutter_localizations/lib/src/l10n/cupertino_es.arb [hunk 6]
```diff
@@ -18,10 +18,10 @@
   "cutButtonLabel": "Cortar",
   "copyButtonLabel": "Copiar",
   "pasteButtonLabel": "Pegar",
-  "selectAllButtonLabel": "Seleccionar todo",
+  "selectAllButtonLabel": "Seleccionar todos",
```
Comment (tgucio) [inline packages/flutter_localizations/lib/src/l10n/cupertino_es.arb:21]: This is incorrect, see https://github.com/flutter/flutter/issues/127871
Same for _es_419.arb etc.

**Review by tgucio — DISMISSED** [review]

**Comment by itsjustkevin** [comment]
> @itsjustkevin What do you think about cherry picking this into stable? Lots of files changed but it's just translations, have we done a translation cherry pick before?
> 
> Currently on stable, iOS users with any text field with spell check enabled will see "No replacements found" in English (untranslated) for words that don't have suggestions. Now that I type that out it seems likely to be a small portion of our users (1. iOS 2. spell check enabled 3. user clicking a word with no suggestions). But people are apparently encountering it since #130874 was filed.

Yes, we are going to have to cherry-pick this into the beta branch.  

I should have caught this one and updated it before the beta branch was cut.  Thanks @thkim1011 and @justinmc

**Comment by justinmc** [comment]
Thanks @hadysata and @tgucio for the translation fixes!  Flutter is lucky to have such an international community.

**Comment by tgucio** [comment]
@justinmc Makes me think if we should maybe mention this PR on Discord to find native volunteers for proofreading in more languages :).

File: packages/flutter_localizations/lib/src/l10n/cupertino_es.arb (hunk 6 above)
Comment (thkim1011) [reply packages/flutter_localizations/lib/src/l10n/cupertino_es.arb:21]: Submitted translation internally.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter_localizations/lib/src/l10n/cupertino_pt.arb (hunk 2 above)
Comment (thkim1011) [reply packages/flutter_localizations/lib/src/l10n/cupertino_pt.arb:21]: I'll address this with https://github.com/flutter/flutter/issues/127871.

File: packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb (hunk 3 above)
Comment (thkim1011) [reply packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb:31]: I'll address this with https://github.com/flutter/flutter/issues/127871.

File: packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb (hunk 4 above)
Comment (thkim1011) [reply packages/flutter_localizations/lib/src/l10n/cupertino_pl.arb:35]: I'll get to this with the other inconsistencies in https://github.com/flutter/flutter/issues/127871.

**Comment by tgucio** [comment]
@thkim1011 In the last commit I can see some messages have been moved around and no longer appear sorted:

```
 {
-  "menuDismissLabel": "Dismiss menu",
   "noSpellCheckReplacementsLabel": "No replacements found",
+  "menuDismissLabel": "Dismiss menu",
   "searchTextFieldPlaceholderLabel": "Search",
   "tabSemanticsLabel": "Tab $tabIndex of $tabCount",
   "datePickerHourSemanticsLabelOne": "$hour o'clock",
```

Shouldn't these messages be alphabetically sorted in a consistent way for all the languages?

**Comment by thkim1011** [comment]
@tgucio They should be added in the same order as `cupertino_en.arb` AFAIK. No guarantees on sorting. All of these are automatically generated.

**Comment by tgucio** [comment]
Thanks @thkim1011.
After merging this one, do we have a rough ETA to fix the translations that will become incorrect again via #127871?

**Pull request merged**